packages/ra-ui-materialui/src/list/datagrid/useDatagridContext.ts

```diff
@@ -0,0 +1,21 @@
+import { useContext, useMemo } from 'react';
+import { DatagridProps } from './Datagrid';
+import DatagridContext, { DatagridContextValue } from './DatagridContext';
+import merge from 'lodash/merge';
+import { warning } from 'ra-core';
+
+export const useDatagridContext = (
+    props?: DatagridProps
+): DatagridContextValue => {
+    const context = useContext(DatagridContext);
+
+    return useMemo(
+        () =>
+            merge(
+                {},
+                context,
```

Comment:
the default context value is `undefined`. I think it's wrong, it should be an empty object.